staging/src/kubevirt.io/api/backup/v1alpha1/types.go

```diff
@@ -49,3 +58,98 @@ type BackupOptions struct {
 	PushPath        *string      `json:"pushPath,omitempty"`
 	SkipQuiesce     bool         `json:"skipQuiesce,omitempty"`
 }
+
+// VirtualMachineBackup defines the operation of backing up a VM
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineBackupSpec `json:"spec"`
+
+	// +optional
+	Status *VirtualMachineBackupStatus `json:"status,omitempty"`
+}
+
+// VirtualMachineBackupList is a list of VirtualMachineBackup resources
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineBackupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// +listType=atomic
+	Items []VirtualMachineBackup `json:"items"`
+}
+
+// VirtualMachineBackupSpec is the spec for a VirtualMachineBackup resource
+type VirtualMachineBackupSpec struct {
+	// +optional
+	// Source specifies the VM to backup
+	// If not provided, a reference to a VirtualMachineBackupTracker must be specified instead
+	Source *corev1.TypedLocalObjectReference `json:"source,omitempty"`
+	// +optional
+	// Mode specifies the way the backup output will be recieved
+	Mode *BackupMode `json:"mode,omitempty"`
+	// +optional
+	// PvcName required in push mode. Specifies the name of the PVC
+	// where the backup output will be stored
+	PvcName *string `json:"pvcName,omitempty"`
```

Comment:
What if a VM has a bunch of disks and full backup can't fit on a single disk? Should the API support multiple PVCs?